### DISCUSSION TITLE
FF to replace Sony MP-F52W drive?

### DISCUSSION THREAD

**Discussion by dl6lr** [keirf/flashfloppy#683] [Q&A] [answered]
Hi Keir,

Thanks for the great software FlashFloppy. If you are interested, it works like a charm in a Gould Logic Analyser K105 with a strange version of CP/M86 and replaces one of the 5,25" DD drives.

I have some vintage Hewlett Packard test equipment. HP used a lot Sony MP-F52W drives. These drives are somehow special: They are DD drives with 1MB unformatted capacity, but they spin at 600rpm (instead of normal 300rpm), so the data rate of MFM encoding is 500Kbit/s (instead of 250KBit/s for normal DD drives).

There is a hint (https://www.rudiswiki.de/wiki9/HxCFloppyDiskEmulator) that HxC-Software can be used, if you have
1. a formatted HP floppy to read in a PC to an HxC image
2. edit the image to indicate 500K data rate.
3. modify the Gotek to use DISKCHNG on pin 2, CHNGRESET on pin 1 and use Shugart interface READY on pin34, DRIVESEL on pin  10 and MOTON on pin16

I think by patching the image HxC emits the correct INDEX pulses, but will not if you try to format a new disk with it.

The HP uses LIF file system on 77 or 79 cylinders with 256Byte or 1024 Byte sectors. Furthermore, the HP uses some kind of "copy protection" on cylinder 78ff or 80 with strange sectors (probably sectors with odd numbers).
 
Problem is, I don't have those floppy, but I have images (IMD and TD0) of the lower 77 cylinders. They are meant to be copied to a disk that is formatted in the HP.

So is there a possibility to emulate such a drive with 600rpm (INDEX pulses), so I can "format" a floppy image in the Gotek with FlashFloppy? If this is possible I can copy the lower cylinders to the images and have functional images. Unfortunately my Sony driv e is borked (and a normal drive will not work due to the wrong speed)...

I am able to assist with Logic Analyzer traces (hopefully, I am just repairing it).

Furthermore: I have one old Gotek left with genuine STM32. Currently it seems I can only get Artery 32F415, not the 2022 version with 32F435. Would this matter in any way?

Regards, Bernd

**Comment by dl6lr** [comment]
Well, to answer my own question: It is possible to emulate the Sony drive. FF.CFG contains:
```
interface=shugart
pin02=chg
pin34=rdy
chgrst=pa14
```
IMG.CFG contains (with the odd sector ids the HP is working with, got these hints from the logging firmware):
```
[4952A]
cyls=80
heads=2
rpm=600
mode=mfm
rate=500
secs=16
bps=256
tracks=0-77
  id=1
tracks=78
  id=17
tracks=79
  id=97
```
Images are 655360 bytes (80*2*16*256) in size.
This is working somehow, but the copy protection scheme of the HP 4952A does not allow formatting of an image and the logging version did not give me any clue what was going wrong. Have to investigate further. Seems slow too, as the manual states formatting takes 45 seconds, but it is way slower, so may have to experiment with all the skew settings.
My Artery based Gotek is too crippled to support the CHGRST on PA14.

**Comment by dl6lr** [comment, marked as answer]
With some help from Keir I was able to get it fully working. The trick was to use HFE files, not RAW images, as the instrument places sector headers with different track ids on the disc. An empty created HFE file (80 cylinders, 2 sides, 16 sectors, 256btes per sector, MFM, data rate 500, rpm 600) can be formatted with the instrument. Images made with Teledisk could be converted to HFE, then patched to the higher data rate. They are working fine.

**Reply by arpruss** [reply]
FlashFloppy is not working for me. I tried the above ff.cfg, and I also tried changing chg and rdy to nchg and nrdy. When the oscilloscope tries to load the system file from an HFE volume, I see all sorts of numbers show up on the display (often going beyond the 0-79 range), and eventually I get a failure message.

I know the HFE volume is good, as the oscilloscope can start from it with HxC.

**Comment by Aruba297** [comment]
I’ve been doing similar with an HP1652B, but using HxC. Didn’t find your post till now. I have a dual drive configuration with the floppy and the Gotek together. When I need to switch I power down the floppy and enable the Gotek. My intent is to boot on the floppy (successfull), and format the Gotek. This completes but checking the Gotek always shows “Disk not supported”. Btw the USB is formatted with the HxC firmware by the push button menu. Any suggestions.

**Reply by arpruss** [reply]
How did you set up the dual floppy configuration?

**Reply by Aruba297** [reply]
See link below for my blog and the info on the Y cable that allows me to select either floppy or Gotek:
https://www.edcheung.com/album/album15/hp1652b/hp1652b.htm#gotek

Video of my going back and forth to copy files:
https://youtu.be/RPMTsWtg6PQ?si=zecYlu6yTyO1MC6Z

Dr. Edward Cheung

**Reply by arpruss** [reply]
Thanks so much! I am guessing that one can't safely hot-swap the gotek and the Sony drive?

Does one need the Y-cable given the .hfe file, or can one just modify a ribbon cable and use the .hfe file to boot the oscilloscope?

**Reply by Aruba297** [reply]
Hm never tried a hot swap.  With all the lines, it may cause a lock up if some wrong sequence of pin mate is performed.  So, I don't know.

Not sure about your second question but let me say that I no longer have the dual boot configuration.  It was only used when I wanted to transfer files from floppy to USB.  Nowadays, I just run completely on the Gotek USB.  So I guess that means the Y-cable is not required by the .hfe file.   But realize that the power configuration of the standard cable is not compatible with the Gotek.  If I recall correctly, plugging the original ribbon into the Gotek will cause a short to the 12V bus.

Dr. Edward Cheung

**Reply by arpruss** [reply]
Thanks again! I haven't opened my oscilloscope yet. Looking at your website, it looks like I need to order a male-to-female 34-pin floppy extender instead of the usual female-to-female one?
And will it be easier to work with a 6-inch or an 18-inch cable?

**Reply by Aruba297** [reply]
The cable needed is shown in my blog referenced above.

**Reply by arpruss** [reply]
It's almost working now. When the oscilloscope boots up it says "no disk" and I need to press a key to try again, and then it boots fine. 

I have an AT3 gotek and I followed the instructions for the AT2.

**Reply by Aruba297** [reply]
I am mystified why it would indicate "no disk" and then be ok later.  Very odd.

**Reply by arpruss** [reply]
I wonder if it has something to do with the Disk Change and Disk Change Reset lines. Following Jeff's instructions for the SFRKC32.AT2, I wired the scope's Disk Change Reset pin 1 line to the Gotek's pin 4, and then connected pin 4 to JK on the board. I wired the scope's Disk Change pin 2 line to the Gotek's pin 2. And I set the firmware to Enable Reset DC (Diskchange) input signal (pin PA14). However, what I have is an SFRKC32.AT3 instead of AT2. Maybe that makes a difference? But the AT3 board looks just like the photos I've seen of the AT2.

**Reply by arpruss** [reply]
Just noticed that the AT3 has a 48-pin Artery and PA14 is not connected to JK. I found a via that is connected to PA14 so I can try to connect that to pin 4 tomorrow if I have time.

**Reply by arpruss** [reply]
It turns out that the AT3 has a PA14 header pin near the front in the same position as the SFRC922 does. Works great now.

**Reply by arpruss** [reply]
Dr Cheung: What FlashFloppy settings are you using, and does FlashFloppy work for you with LIF images, or do you have to use HFE?

**Reply by Aruba297** [reply]
Here is the blog and my notes on this project:
https://www.edcheung.com/album/album15/hp1652b/hp1652b.htm

It has a file with the firmware used on my Gotek.  If you have questions after checking that out, post a reply.
